File: app/src/main/java/ostmodern/skylark/di/ActivityBuilderModule.java
package ostmodern.skylark.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ostmodern.skylark.MainActivity;

@Module
public abstract class ActivityBuilderModule {

    @ContributesAndroidInjector(modules = FragmentBuildersModule.class)
    abstract MainActivity contributeMainActivity();

}

File: app/src/main/java/ostmodern/skylark/di/ActivityBuilderModule.kt
package ostmodern.skylark.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import ostmodern.skylark.MainActivity

@Module
abstract class ActivityBuilderModule {

    @ContributesAndroidInjector(modules = [(FragmentBuildersModule::class)])
    internal abstract fun contributeMainActivity(): MainActivity
}
